web/src/message_scroll.js

```diff
@@ -87,6 +87,16 @@ export function scroll_finished() {
 
     if (message_scroll_state.update_selection_on_next_scroll) {
         message_viewport.keep_pointer_in_view();
+        // If we don't want to update message selection on this scroll,
+        // we also don't want to mark any visible messages as read and
+        // are waiting on user input to do so. So, we only mark messages
+        // as read if we are updating selection on this scroll.
+        //
+        // When the window scrolls, it may cause some messages to
+        // enter the screen and become read.  Calling
+        // unread_ops.process_visible will update necessary
+        // data structures and DOM elements.
+        setTimeout(unread_ops.process_visible, 0);
```

Comment:
Why does this have a `setTimeout` on it? `scroll_finished` is already debounced/delayed, and checks if we're even in the same view we were scrolling... but I'm just feeling a bit uncertain given that `message_viewport.keep_pointer_in_view` should have updated the scroll position already.

Likely not a regression, given the previous code.

Reply:
Opened #32099 for that.